server/endpoint/endpoint.go

```diff
@@ -16,9 +19,36 @@ type Config struct {
 
 // New creates a new endpoint with given configuration.
 func New(config Config) (*Endpoint, error) {
-	return &Endpoint{}, nil
+	var err error
+
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+	}
+
+	if config.Service == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Service or it's Healthz descendents must not be empty")
+	}
+
+	var healthzEndpoint *healthz.Endpoint
+	{
+		c := healthz.DefaultConfig()
+		c.Logger = config.Logger
+		c.Services = []healthzservice.Service{
+			config.Service.Healthz.K8s,
+		}
+
+		healthzEndpoint, err = healthz.New(c)
+		if err != nil {
+			return nil, microerror.Maskf(err, "healthz.New")
+		}
+	}
+
+	return &Endpoint{
+		Healthz: healthzEndpoint,
+	}, nil
```

Comment:
I would like to keep it explicit. Returning nil is somehow hidden and mixed with the construction. In golang there is a primer that says to rather be explicit than smart. It makes life easier.

Reply:
Did you not agree or not notice this here? I should not approve next time when there are things left.

Reply:
I agreed with you, but misunderstood. Did you mean this to be like this:
```
endpoint := &Endpoint {
    Healthz: healthzEndpoint,
}

return endpoint, nil
```

or what did you mean?

Reply:
I mean this. This would be cool to have. It went unchanged into master though. See https://github.com/giantswarm/cluster-operator/blob/138f5cce71f53718c8a1e2e56c89031e69f912a6/server/endpoint/endpoint.go#L46-L48.